index.js

```diff
@@ -353,14 +356,14 @@ PersonaClient.prototype.validateToken = function (opts, next) {
                 return next(ERROR_TYPES.VALIDATION_FAILURE, null, decodedToken);
             }
 
-            if(scope && decodedToken.hasOwnProperty("scopeCount")) {
+            if(scopes && decodedToken.hasOwnProperty("scopeCount")) {
                 debug("Token has too many scopes (" + decodedToken.scopeCount + ") to put in payload, asking Persona...");
-                return headScopeThenVerify(scope, next, decodedToken);
-            } else if (scope == null || _.includes(decodedToken.scopes, "su") || _.includes(decodedToken.scopes, scope)) {
+                return headScopeThenVerify(next, decodedToken);
+            } else if (scopes == null || _.includes(decodedToken.scopes, "su") || _.intersection(decodedToken.scopes, scopes).length > 0) {
```

Comment:
this condition is checking for several different things and it might not be obvious, on first reading, what these represent, could you add a comment here explaining these clauses.